google-cloud-spanner/lib/google/cloud/spanner/service.rb

```diff
@@ -408,6 +416,18 @@ def inspect
 
         protected
 
+        def service_address
+          return nil if host.nil?
+          host.split(":").first
+        end
+
+        def service_port
+          return nil if host.nil?
+          port = host.split(":")[1]
+          return nil if port.nil?
+          port.to_i
```

Comment:
Is this conversion to Integer adequate without additional error handling and messaging?

Reply:
This will convert non-numeric strings to 0, which will silently swallow the error and may not be what is wanted.

Reply:
What do you think about leaning on URI here?

```
irb(main):021:0> u = URI "https://localhost:nan"
URI::InvalidURIError: bad URI(is not URI?): https://localhost:nan
	from /Users/quartzmo/.rbenv/versions/2.4.5/lib/ruby/2.4.0/uri/rfc3986_parser.rb:67:in `split'
	from /Users/quartzmo/.rbenv/versions/2.4.5/lib/ruby/2.4.0/uri/rfc3986_parser.rb:73:in `parse'
	from /Users/quartzmo/.rbenv/versions/2.4.5/lib/ruby/2.4.0/uri/common.rb:231:in `parse'
	from /Users/quartzmo/.rbenv/versions/2.4.5/lib/ruby/2.4.0/uri/common.rb:737:in `URI'
	from (irb):21
	from /Users/quartzmo/.rbenv/versions/2.4.5/bin/irb:11:in `<main>'
```

Reply:
Success path:

```
irb(main):022:0> host = "localhost:1234"
=> "localhost:1234"
irb(main):023:0> u = URI "https://#{host}"
=> #<URI::HTTPS https://localhost:1234>
irb(main):024:0> u.host
=> "localhost"
irb(main):025:0> u.port
=> 1234
```

Reply:
As long as port defaults to `nil` (rather than, e.g., 80) if it's omitted.

Reply:
```ruby
host = "localhost"

require "uri"
uri = URI.parse "//#{host}"
uri.host #=> "localhost"
uri.port #=> nil
```

Reply:
```ruby
host = "localhost:1234"

require "uri"
uri = URI.parse "//#{host}"
uri.host #=> "localhost"
uri.port #=> 1234
```

Reply:
@blowmage, That looks good, i'll omit scheme as you showed above, great idea.

Reply:
Leaning on URI is smart. Good job!